okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -321,11 +320,12 @@ public void transportHeadersReceived(List<Header> headers, boolean endOfStream)
      * Must be called with holding the transport lock.
      */
     @GuardedBy("lock")
-    public void transportDataReceived(okio.Buffer frame, boolean endOfStream) {
+    public void transportDataReceived(okio.Buffer frame, boolean endOfStream, int paddingLen) {
       // We only support 16 KiB frames, and the max permitted in HTTP/2 is 16 MiB. This is verified
       // in OkHttp's Http2 deframer. In addition, this code is after the data has been read.
       int length = (int) frame.size();
       window -= length;
+      processedWindow -= (length + paddingLen);
```

Comment:
This should be:
```java
      window -= length + paddingLen;
      processedWindow -= paddingLen;
```

And we shouldn't need to change `bytesRead()`. We want to delay "returning" bytes to processedWindow, as that delays when (and how many bytes) we'll do the WINDOW_UPDATE.